strings/base_error.h

```diff
@@ -546,6 +546,28 @@ namespace winrt::impl
         }
         return result;
     }
+
+    template <typename T>
+    WINRT_IMPL_NOINLINE void check_cast_result(T* from WINRT_IMPL_SOURCE_LOCATION_ARGS)
```

Comment:
Doesn't this have to be `WINRT_IMPL_NOINLINE inline` to avoid ODR problems?

Reply:
What sort of ODR problems?  The only variance based on compiler options is the SOURCE_LOCATION macro and that already has global ODR guards.

The `noinline` is important because it is how the code bloat is avoided by having lots of exception throwing locations spewed out into containing code.